apps/frontend/src/pages/EditListing.tsx

```diff
@@ -1,5 +1,422 @@
+                          setImageBinaries={setListingImageBinaries}
+                        />
+                      </Box>
+                      <Box>
+                        <Button
+                          sx={{
+                            display: "inline",
+                            mt: 2,
+                            backgroundColor: colors.martletplaceRed,
+                            "&:hover": {
+                              backgroundColor: colors.martletplaceRedHover,
+                            },
+                            textTransform: "none",
+                            fontSize: "16px",
+                            padding: "10px 20px",
+                            margin: "10px",
+                          }}
+                          variant="contained"
+                          id="delete-button"
+                          onClick={handleDelete}
+                        >
+                          Delete Posting
+                        </Button>
+                        <Button
+                          sx={{
+                            display: "inline",
+                            mt: 2,
+                            backgroundColor: colors.martletplaceYellow,
+                            "&:hover": {
+                              backgroundColor: colors.martletplaceYellowHover,
+                            },
+                            textTransform: "none",
+                            fontSize: "16px",
+                            padding: "10px 20px",
+                            margin: "10px",
+                          }}
+                          variant="contained"
+                          id="status-button"
+                          onClick={handleUpdateStatus}
+                        >
+                          {newListingObject.listing.status === "AVAILABLE"
+                            ? "Mark Purchased"
+                            : "Mark Not Purchased"}
+                        </Button>
+                      </Box>
+                    </Box>
+                  </form>
+                </Box>
+              </Grid>
+              <Grid item lg={6} xs={12}>
+                <Box>
+                  <Typography variant="h5" sx={{ paddingLeft: "20px" }}>
+                    Image Preview
+                  </Typography>
+                  <Box sx={{ padding: "10px" }}>
+                    {!isImageValid(listingImages[0]) ? (
+                      <Typography sx={{ paddingLeft: "10px" }} variant="body2">
+                        No images uploaded yet, these images will overwrite the
+                        current ones.
+                      </Typography>
+                    ) : (
+                      <Carousel imageURLs={listingImages} />
+                    )}
+                  </Box>
+                </Box>
+              </Grid>
+            </Grid>
+          </CardContent>
+        </Card>
+      ) : (
+        <Typography>Listing Not Valid</Typography>
```

Comment:
I'm a bit confused when we're showing this and wondering if the text sufficienly explains what's going on?

Do we want to show `Listing Not Valid` when we can't edit the listing or it doesn't work?

Reply:
Yeah, it's kinda placeholder text for if the listing can't be received/doesn't exist, that way they won't see the edit fields if for some reason the listing can't be edited.

Reply:
Maybe something like `Sorry, we were unable to get this listing. Please try again or contact for support.`

Reply:
Love it, I'll change that rn!